src/hotspot/share/cds/filemap.cpp

```diff
@@ -842,6 +885,68 @@ bool FileMapInfo::classpath_failure(const char* msg, const char* name) {
   return false;
 }
 
+char* FileMapInfo::longest_common_app_classpath(int num_paths,
+                                                GrowableArray<const char*>* rp_array) {
+  // find the shortest path length among all the entries
+  char* chr = strrchr((char*)rp_array->at(0), *os::file_separator());
+  size_t min_len = chr != NULL ? chr - rp_array->at(0) + 1 : strlen(rp_array->at(0)) + 1;
+  for (int i = 1; i < num_paths; i++) {
+    chr = strrchr((char*)rp_array->at(i), *os::file_separator());
+    size_t curr_len = chr != NULL ? chr - rp_array->at(i) + 1 : strlen(rp_array->at(i)) + 1;
```

Comment:
If `chr == NULL`, it means we have a file name only, with no file separator. In this case, there will be no common prefix, so this function can return NULL immediately.

Reply:
Right. I don't need to use strrchr in my updated change.